content/kb/tutorials/databases/gcs.md

```diff
@@ -122,35 +124,19 @@ Copy the code below to your Streamlit app and run it. Make sure to adapt the nam
 # streamlit_app.py
 
 import streamlit as st
-from google.oauth2 import service_account
-from google.cloud import storage
+from st_files_connection import FilesConnection
 
-# Create API client.
-credentials = service_account.Credentials.from_service_account_info(
-    st.secrets["gcp_service_account"]
-)
-client = storage.Client(credentials=credentials)
-
-# Retrieve file contents.
-# Uses st.cache_data to only rerun when the query changes or after 10 min.
-@st.cache_data(ttl=600)
-def read_file(bucket_name, file_path):
-    bucket = client.bucket(bucket_name)
-    content = bucket.blob(file_path).download_as_string().decode("utf-8")
-    return content
-
-bucket_name = "streamlit-bucket"
-file_path = "myfile.csv"
-
-content = read_file(bucket_name, file_path)
+# Create connection object and retrieve file contents.
+# Specify input format is a csv and to cache the result for 600 seconds.
+conn = st.experimental_connection('gcs', type=FilesConnection)
+df = conn.read("streamlit-bucket/myfile.csv", input_format="csv", ttl=600)
 
 # Print results.
-for line in content.strip().split("\n"):
-    name, pet = line.split(",")
-    st.write(f"{name} has a :{pet}:")
+for row in df.itertuples():
+    st.write(f"{row.Owner} has a :{row.Pet}:")
 ```
 
-See `st.cache_data` above? Without it, Streamlit would run the query every time the app reruns (e.g. on a widget interaction). With `st.cache_data`, it only runs when the query changes or after 10 minutes (that's what `ttl` is for). Watch out: If your database updates more frequently, you should adapt `ttl` or remove caching so viewers always see the latest data. Learn more in [Caching](/library/advanced-features/caching).
+See `st.experimental_connection` above? This handles secrets retrieval, setup, result caching and retries. By default, `read()` results are cached without expiring. In this case, we set `ttl=600` to ensure the file contents is cached for no longer than 10 minutes. You can also set `ttl=None` to disable caching. Learn more in [Caching](/library/advanced-features/caching).
```

Comment:
```suggestion
See `st.experimental_connection` above? This handles secrets retrieval, setup, result caching and retries. By default, `read()` results are cached without expiring. In this case, we set `ttl=600` to ensure the file contents is cached for no longer than 10 minutes. You can also set `ttl=0` to disable caching. Learn more in [Caching](/library/advanced-features/caching).
```

According to https://docs.streamlit.io/library/api-reference/connections/st.experimental_connection, `ttl=None` means the cache doesn't expire, so `ttl=0` should actually disable the cache.